scripts/load_testing/locustfile.py

```diff
@@ -320,46 +412,190 @@ def sp_rails_change_pass(self):
-        resp = self.client.get('https://www.test.usajobs.gov/Applicant/ProfileDashboard/Home')
-        resp.raise_for_status()
-        # we should now have been redirected to
-        # https://login.test.usajobs.gov/Access/Transition
-        # we could put a resp.url check in here to verify that
-        # We'll now navigate into the regular IDP login flow.
+        root_url = 'https://www.test.usajobs.gov'
+        with self.client.get(root_url, catch_response=True) as resp:
+            dom = resp_to_dom(resp)
+            signin_link = dom.find(
+                'a.user-logged-out[href="/Applicant/ProfileDashboard/Home"]'
+            ).eq(0)
+
+            if not signin_link:
+                resp.failure(
+                    "We could not find a signin link at {}".format(resp.url))
+
+        transition_resp = self.client.get(
+            root_url + signin_link[0].attrib['href'],
+            catch_response=True
+        )
+        with transition_resp as resp:
+            if resp.status_code is not 200:
+                resp.failure(
+                    "Bad {} response at {} with headers {}. Response:".format(
+                        resp.status_code, resp.url, resp.headers, resp.content
+                    )
+                )
+            # we should have been redirected to
+            # https://login.test.usajobs.gov/Access/Transition.
+            # Let's do a quick check.
+            if "login.test.usajobs.gov/Access/Transition" not in resp.url:
+                resp.failure(
+                    """"
+                    We do not appear to have been redirected to
+                    https://login.test.usajobs.gov/Access/Transition.
+                    Our current URL is {}, with content {}.
+                    """.format(resp.url, resp.content)
+                )
+                return
+        # Now that we've confirmed we're at the right URL
+        # we need to POST to it, per USAjobs.
+        handshake_post = self.client.post(
+            resp.url,
+            data={},
+            name="/Access/Transition",
+            catch_response=True
+        )
+        with handshake_post as resp:
+            # Check to make sure we redirected to our target host,
+            # with a request_id in resp.url
+            if resp.url is not os.getenv('TARGET_HOST'):
+                resp.failure(
+                    """"
+                    We do not appear to have been redirected to the IDP host.
+                    Instead, we are at {}.
+                    """.format(resp.url)
+                )
+                return
         credentials = random_cred()
         login(self, credentials)
         change_pass(self, "thisisanewpass")
-        # now change it back.
         change_pass(self, credentials['password'])
         logout(self)
 
-    @locust.task(2)
-    def idp_create_account(self):
-        print("Task: Create account from idp")
-        signup(self)
+    #@locust.task(70)
```

Comment:
Is this commented out on purpose?

Reply:
Yes. Until we have whitelisted IPs for USAGov, those tasks just throw errors. The task is sound -- at least we think it is -- but attempting to run it will start popping 503s and other errors. By commenting out the decorator, the task isn't run.